p2p/libp2p/networksharding/listKadSharder.go

```diff
@@ -0,0 +1,164 @@
+package networksharding
+
+import (
+	"fmt"
+	"math/big"
+	"sort"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+const minAllowedConnectedPeers = 2
+const minAllowedPeersOnList = 1
+
+// listKadSharder is the struct able to compute an eviction list of connected peers id according to the
+// provided parameters. It basically splits all connected peers into 3 lists: intra shard peers, cross shard peers
+// and unknown peers by the following rule: both intra shard and cross shard lists are upper bounded to provided
+// maximum levels, unknown list is able to fill the gap until maximum peer count value is fulfilled.
+type listKadSharder struct {
```

Comment:
There are 2 Sharder Interfaces, one that is implemented in the listKadSharder and the other in the kadSharder. There was a TODO somewhere for this.

Reply:
Unfortunately, interfaces do not match and because how the first implementation was done, are incompatible. Will try to come up with something that works and have only one implementation.

Reply:
ok

Reply:
ok